src-java/blue-green/src/main/java/org/openkilda/bluegreen/ZkWatchDog.java

```diff
@@ -169,4 +171,24 @@ protected void notifyBuildVersionObservers() {
             observer.handle(buildVersion);
         }
     }
+
+    @Override
+    public void close() {
+        if (!observers.isEmpty()) {
+            log.error("Could not close connection to zookeeper {}. Watch dog still have {} life cycle observers",
+                    connectionString, observers.size());
+            return;
```

Comment:
contract violation - the caller will never know that `close()` request has failed and will unable to make more close attempts.

Reply:
fixed